Vonk.Plugin.DocumentOperation/DocumentService.cs

```diff
@@ -228,7 +228,10 @@ private async Task<(bool success, Resource resolvedResource, string failedRefere
         {
             var result = await _searchRepository.GetByKey(ResourceKey.Parse(reference));
             if (result == null)
+            {
+                _missingReferenceIssue = ReferenceNotResolvedIssue(reference, true);
```

Comment:
Communication between method through private fields on the class is errorprone. What if the same service is called twice in the same scope?
Either pass the response to the method to add the issue to the response.Outcome directly or return the issue to the caller.